src/node_file.cc

```diff
@@ -111,6 +112,44 @@ using v8::Value;
 #endif
 
 #define GET_OFFSET(a) ((a)->IsNumber() ? (a).As<Integer>()->Value() : -1)
+#define GET_TRACE_ENABLED                                                  \
+  (*TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED                             \
+  (TRACING_CATEGORY_NODE2(fs, sync)) != 0)
+#define FS_SYNC_TRACE_BEGIN(syscall, ...)                                  \
+  if (GET_TRACE_ENABLED)                                                   \
+  TRACE_EVENT_BEGIN(TRACING_CATEGORY_NODE2(fs, sync), syscall, ##__VA_ARGS__);
+#define FS_SYNC_TRACE_END(syscall, ...)                                    \
+  if (GET_TRACE_ENABLED)                                                   \
+  TRACE_EVENT_END(TRACING_CATEGORY_NODE2(fs, sync), syscall, ##__VA_ARGS__);
+
+static const char* SYNC_TRACE_ACCESS = "fs.access";
+static const char* SYNC_TRACE_CHMOD = "fs.chmod";
+static const char* SYNC_TRACE_CHOWN = "fs.chown";
+static const char* SYNC_TRACE_CLOSE = "fs.close";
+static const char* SYNC_TRACE_COPYFILE = "fs.copyfile";
+static const char* SYNC_TRACE_FCHMOD = "fs.fchmod";
+static const char* SYNC_TRACE_FCHOWN = "fs.fchown";
+static const char* SYNC_TRACE_FDATASYNC = "fs.fdatasync";
+static const char* SYNC_TRACE_FSTAT = "fs.fstat";
+static const char* SYNC_TRACE_FSYNC = "fs.fsync";
+static const char* SYNC_TRACE_FTRUNCATE = "fs.ftruncate";
+static const char* SYNC_TRACE_FUTIME = "fs.futimes";
+static const char* SYNC_TRACE_LINK = "fs.link";
+static const char* SYNC_TRACE_LSTAT = "fs.lstat";
+static const char* SYNC_TRACE_MKDIR = "fs.mkdir";
+static const char* SYNC_TRACE_MKDTEMP = "fs.mkdtemp";
+static const char* SYNC_TRACE_OPEN = "fs.open";
+static const char* SYNC_TRACE_READ = "fs.read";
+static const char* SYNC_TRACE_READDIR = "fs.readdir";
+static const char* SYNC_TRACE_READLINK = "fs.readlink";
+static const char* SYNC_TRACE_REALPATH = "fs.realpath";
+static const char* SYNC_TRACE_RENAME = "fs.rename";
+static const char* SYNC_TRACE_RMDIR = "fs.rmdir";
+static const char* SYNC_TRACE_STAT = "fs.stat";
+static const char* SYNC_TRACE_SYMLINK = "fs.symlink";
+static const char* SYNC_TRACE_UNLINK = "fs.unlink";
+static const char* SYNC_TRACE_UTIME = "fs.utimes";
+static const char* SYNC_TRACE_WRITE = "fs.write";
```

Comment:
I'd prefer to not have to define these as static strings this way and just build it in to the macro such that, for instance, `FS_SYNC_TRACE_BEGIN(access)` would expand out to `fs.access`